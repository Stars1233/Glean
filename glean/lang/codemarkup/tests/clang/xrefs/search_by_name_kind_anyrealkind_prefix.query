query: search.code.SearchByNameKindAndLanguage { name = "f".., language = Cpp, kinds = { just = _ } }
perf: true
transform: [gensort, []]
