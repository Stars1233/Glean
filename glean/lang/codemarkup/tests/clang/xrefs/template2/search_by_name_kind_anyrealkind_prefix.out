[
  "@generated",
  {
    "key": {
      "name": "f",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "decl": {
            "function_": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": { "name": { "key": "f" } } },
                    "scope": {
                      "recordWithAccess": {
                        "record": {
                          "key": {
                            "name": { "key": "A" },
                            "scope": { "global_": { } }
                          }
                        },
                        "access": 0
                      }
                    }
                  }
                },
                "signature": {
                  "key": {
                    "returns": { "key": "void" },
                    "parameters": [
                      { "name": { "key": "" }, "type": { "key": "U" } }
                    ]
                  }
                },
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 10,
                  "columnBegin": 24,
                  "lineEnd": 10,
                  "columnEnd": 39
                }
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "name": "f",
      "language": 0,
      "kinds": 13,
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "f" } } },
                        "scope": {
                          "recordWithAccess": {
                            "record": {
                              "key": {
                                "name": { "key": "A" },
                                "scope": { "global_": { } }
                              }
                            },
                            "access": 0
                          }
                        }
                      }
                    },
                    "signature": {
                      "key": {
                        "returns": { "key": "void" },
                        "parameters": [
                          { "name": { "key": "" }, "type": { "key": "int" } }
                        ]
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 13,
                      "columnBegin": 1,
                      "lineEnd": 15,
                      "columnEnd": 28
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      }
    }
  }
]